variables.tf

```diff
@@ -0,0 +1,290 @@
+
+variable "zone_name" {
+  type        = string
+  description = "Name of the private dns zone"
+  default     = "privatelink.vaultcore.azure.net"
+}
+
+variable "soa_record" {
+  type = object({
+    email        = string
+    expire_time  = number
+    minimum_ttl  = number
+    refresh_time = number
+    retry_time   = number
+    ttl          = number
+    tags         = map(string)
+  })
+  default = null
+}
+
+variable "private_dns_zone_tags" {
+  description = "Map of tags to be associated with the resource"
+  type        = map(string)
+  default     = {}
+}
+
+################################################
+# Variables related to private DNS zone link
+################################################
+
+variable "virtual_network_id" {
+  description = "The ID of the Virtual Network that should be linked to the DNS Zone. Changing this forces a new resource to be created."
+  type        = string
+}
+
+variable "registration_enabled" {
+  description = "(Optional) Is auto-registration of virtual machine records in the virtual network in the Private DNS zone enabled? Defaults to false."
+  type        = bool
+  default     = false
+}
+
+variable "private_dns_zone_link_vnet_tags" {
+  description = "Tags to be associated with this resource"
+  type        = map(string)
+  default     = {}
+}
+
+################################################
+# Variables related to private endpoint
+################################################
+
+variable "subnet_id" {
+  description = <<EOT
+    The ID of the Subnet from which Private IP Addresses will be allocated for this Private Endpoint.
+    Changing this forces a new resource to be created.
+  EOT
+  type        = string
+}
+
+variable "private_dns_zone_group_name" {
+  description = "Specifies the Name of the Private DNS Zone Group."
+  type        = string
+  default     = ""
+}
+
+variable "is_manual_connection" {
+  description = <<EOT
+    Does the Private Endpoint require Manual Approval from the remote resource owner? Changing this forces a new resource
+    to be created.
+  EOT
+  type        = bool
+  default     = false
+}
+
+variable "subresource_names" {
+  description = <<EOT
+    A list of subresource names which the Private Endpoint is able to connect to. subresource_names corresponds to group_id.
+    Possible values are detailed in the product documentation in the Subresources column.
+    https://docs.microsoft.com/azure/private-link/private-endpoint-overview#private-link-resource
+  EOT
+  type        = list(string)
+  default     = ["vault"]
+}
+
+variable "request_message" {
+  description = <<EOT
+    A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection
+    to the remote resource. The request message can be a maximum of 140 characters in length.
+    Only valid if `is_manual_connection=true`
+  EOT
+  type        = string
+  default     = ""
+}
+
+variable "private_endpoint_tags" {
```

Comment:
We should only have one set of tags named `tags` those should be passed in to all the modules that are wrapped in this ref arch. Since all these resources which are part of this module will have the same lifecycle I dont see a reason why we should have different tags for each module.